### ISSUE TITLE
Add WebHook middleware for existing connect-comptaible HTTP servers (e.g. express)

### ISSUE THREAD

**Issue by d-fischer** [d-fischer/twitch#111]
## Feature

Instead of `listener.listen()`, you should be able to do something like:
```ts
const app = express();
const listener = await WebHookListener.create(/* ... */);
listener.applyMiddleware(app);
```

**Comment by thomaslule** [comment]
Yes! Or `app.use('/webhooks', listener.asExpressMiddleware());`

**Comment by d-fischer** [comment]
> `app.use('/webhooks', listener.asExpressMiddleware());`

I specifically avoided writing this as an example because the listener should already know the base URL. 

For some reason, I still included it in the params of the example call initially. I just edited the example accordingly.

**Comment by thomaslule** [comment]
OK if I understand correctly you prefer a "smart" listener that knows the reverse-proxy configuration and from this configuration, it calculates the callback URL to communicate to twitch and the base URL to listen to.

I was imagining a "dumb" listener that only knows the public callback URL, it doesn't know the base path where it is mounted, it only transmits the callback URL to twitch without even reading it and answers to the http requests that reach it.

At first sight, the "dumb" solution seems easier to implement and easier to use but I guess I'm missing something?

**Comment by Satont** [comment]
This is the best, prefered for the most users way.

Cause it's easiest way.

**Comment by d-fischer** [comment]
This was released earlier with 4.1.0-pre.11.
